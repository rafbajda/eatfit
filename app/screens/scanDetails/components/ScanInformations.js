import React from 'react';
import hps from '../utils/helpers';
import {
    ScanInformationContainer, ScanInformationDescription, ScanInformationInfo,
    ScanInformationLabel, ScanRate, ScanRatingIcon
} from '../styles/scanDetailsStyles';

const ScanInformations = props => {
    const { score } = props;
    const roundedScore = Math.round(score * 100) / 100;
    const emoji = hps.getEmoji(score);
    const color = hps.getColor(score);
    const description = hps.getDescription(score);
    return (
        <ScanInformationContainer>
            <ScanInformationLabel>Rating:</ScanInformationLabel>
            <ScanInformationInfo>
                <ScanRate color={color}>{roundedScore}</ScanRate>
                <ScanRatingIcon>{emoji}</ScanRatingIcon>
            </ScanInformationInfo>
            <ScanInformationDescription color={color}>
                {description}
            </ScanInformationDescription>
        </ScanInformationContainer>
    );
};

export default ScanInformations;
