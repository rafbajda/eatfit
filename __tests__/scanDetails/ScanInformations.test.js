import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import ScanInformations from '../../app/screens/scanDetails/components/ScanInformations';

describe('ScanImage snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<ScanInformations />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
