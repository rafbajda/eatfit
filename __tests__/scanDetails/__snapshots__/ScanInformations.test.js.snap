// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanImage snapshot renders component 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  className="sc-bwzfXH uTxCW"
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": "10%",
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text
      className="sc-htpNat bRZdru"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      [missing "en.labels.rating" translation]
    </Text>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      className="sc-bxivhb bUZGRr"
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <Text
          className="sc-EHOje kqRTHu"
          color="#00FF00"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          NaN
        </Text>
        <Text
          className="sc-ifAKCX eBNren"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          😁
        </Text>
      </View>
    </RCTScrollView>
    <Text
      className="sc-bZQynM iQNpmG"
      color="#00FF00"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      [missing "en.healthImpactLabels.veryGood" translation]
    </Text>
  </View>
</RCTScrollView>
`;
